src/Sarif/FileRegionsCache.cs

```diff
@@ -1,23 +1,35 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
 using System;
 using System.IO;
 
 namespace Microsoft.CodeAnalysis.Sarif
 {
     /// <summary>
-    /// This class is a file cache that can be used to populate
-    /// regions with comprehensive data, to retrieve file text
-    /// associated with a SARIF log, and to construct text
-    /// snippets associated with region instances.
+    /// A cache that can be used to populate regions with comprehensive data, to retrieve artifact
+    /// text associated with a SARIF log, and to construct snippets for <see cref="Region"/>
+    /// instances.
     /// </summary>
-    public class FileRegionsCache
+    public class FileRegionsCache : IFileRegionsCache
     {
         public const int DefaultCacheCapacity = 100;
         private readonly Run _run;
         private readonly IFileSystem _fileSystem;
         private readonly Cache<string, Tuple<string, NewLineIndex>> _cache;
 
+        /// <summary>
+        /// Creates a new <see cref="FileRegionsCache"/> object.
+        /// </summary>
+        /// <param name="run">
+        /// The <see cref="Run"/> object whose data is to be cached.
+        /// </param>
+        /// <param name="capacity">
+        /// The initial capacity of the cache.
+        /// </param>
+        /// <param name="fileSystem">
+        /// An object that provides access to file system services.
+        /// </param>
         public FileRegionsCache(Run run, int capacity = DefaultCacheCapacity, IFileSystem fileSystem = null)
```

Comment:
>capacity [](start = 45, length = 8)

By the way, I noticed that the parameter `capacity` is unused. It should have been passed as the second argument to the `Cache` ctor just below.

Reply:
just pushed a change fixing it.

---
In reply to: [511647723](https://github.com/microsoft/sarif-sdk/pull/2130#discussion_r511647723) [](ancestors = 511647723)